plugins/colorbutton/plugin.js

```diff
@@ -370,6 +414,152 @@ CKEDITOR.plugins.add( 'colorbutton', {
 			}
 		}
 
+		function renderContentColors( options ) {
+			// This function is called on each dialog opening.
+			var contentColorsRow = options.contentColorsRow,
+				contentColorsLabel = options.contentColorsLabel,
+				cssAttribute = options.cssAttribute,
+				clickFn = options.clickFn,
+				type = options.type,
+				colorSpans = editor.editable().find( 'span[style*=' + cssAttribute + ']' ).toArray(),
+				colors = [],
+				colorOccurrences = {},
+				sortedColors = [],
+				colorsPerRow = config.colorButton_colorsPerRow || 6,
+				tilePosition = 1;
+
+			// It's more reliable and easier to delete exisiting color tiles and recreate them than to manipulate the old ones.
+			CKEDITOR.tools.array.forEach( contentColorsRow.find( 'td' ).toArray(), function( node ) {
+				node.remove();
+			}, 0, 1 );
+
+			if ( !colorSpans.length ) {
+				contentColorsLabel.hide();
+				return;
+			}
+
+			CKEDITOR.tools.array.forEach( colorSpans, function( span ) {
+				// colorSpans may contain spans with background color when they are not necessary, so here they are being filtered out.
+				var spanColor = span.getStyle( cssAttribute );
+
+				if ( spanColor ) {
+					colors.push( normalizeColor( spanColor ).toUpperCase() );
+				}
+			} );
+
+			if ( !colors.length ) {
+				contentColorsLabel.hide();
+				return;
+			}
+
+			CKEDITOR.tools.array.forEach( colors, function( color ) {
+				if ( color in colorOccurrences ) {
+					colorOccurrences[ color ]++;
```

Comment:
It's a bit easier to read something like `colorOccurences[ color ] += 1` or even `colorOccurences[ color ] = colorOccurences[ color ] + 1` than incrementation. Overall, incrementation works fine mostly for step variables like `i++` in for loop, otherwise, it's a bit unreadable and sometimes ends up in unexpected behavior, like legendary
```
y = 0
x = y++
x == 1 // Oh snap ;( we should use ++y instead
```